### PR TITLE
Reduce the space above the tabstrip in Windows

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/35972

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-index bd59ccd1b178450c7eba720ea8b235f24416a15e..3b2be5646116d60f79baf3e968aa164245dde937 100644
+index bd59ccd1b178450c7eba720ea8b235f24416a15e..40d21c79c71f8fb76dcaa8910ebd011db0443d19 100644
 --- a/chrome/browser/ui/views/frame/browser_frame_view_win.cc
 +++ b/chrome/browser/ui/views/frame/browser_frame_view_win.cc
-@@ -547,6 +547,7 @@ int BrowserFrameViewWin::TopAreaHeight(bool restored) const {
+@@ -488,6 +488,7 @@ int BrowserFrameViewWin::FrameTopBorderThickness(bool restored) const {
+       // default. When maximized, the OS sizes the window such that the border
+       // extends beyond the screen edges. In that case, we must return the
+       // default value.
++      BRAVE_BROWSER_FRAME_VIEW_WIN_FRAME_TOP_BORDER_THICKNESS
```
Comment (mkarolin) [inline patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch:9]: Any chance we can subclass `BrowserFrameViewWin`, make `FrameTopBorderThickness` and override it?

Comment (zenparsing) [reply patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch:9]: I'll look into it. The existing patch in this file does basically the same thing that this one does: mimics the early return that upstream has for its 2023 refresh. We should probably change techniques for both at the same time. Would it be OK to go with this approach for now (we'll want to uplift) and create an issue to explore a subclassing approach?

Comment (mkarolin) [reply patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch:9]: Sure, let's create a follow up issue, if you don't mind.

Comment (zenparsing) [reply patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch:9]: @emerick has pointed out that a subclass already exists, so I'll take another look at changing the strategy in this PR.

Comment (zenparsing) [reply patches/chrome-browser-ui-views-frame-browser_frame_view_win.cc.patch:9]: https://github.com/brave/brave-browser/issues/35980

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.65.23 Chromium: 122.0.6261.57 (Official Build) nightly (64-bit)
-- | --
Revision | f279160ea3cd8601d5c37a754cd71140c3594e9e
OS | Windows 11 Version 23H2 (Build 22631.3155)
```

Using the STR/Examples outlined via https://github.com/brave/brave-browser/issues/35972#issue-2130518393, ensured that the spacing for horizontal tabs was reduced as per the following:

**`1.64.58 Chromium: 121.0.6167.164`** | **`1.65.23 Chromium: 122.0.6261.57`**
--------------------------------------------|--------------------------------------------
spacingReproduced | spacingFixed

**Pull request merged**